jmsSource/REAMDE.md

```diff
@@ -0,0 +1,321 @@
+from which to read. The following is an example of a Procedure created in VANTIQ Modelo querying against a JMS Source:
+
+```
+PROCEDURE readMessageFromQueue()
+
+SELECT * FROM SOURCE JMS1 AS msg WITH
+	operation: "read",
+	queue: "NamirJMSServer-0/NamirSystemModule-0!NamirJMSServer-0@/com/namir/weblogic/base/dq"
+  
+  {
+      var myObj = {}
+      myObj.JMSMessage = msg.message
+      myObj.JMSFormat = msg.JMSFormat
+      myObj.JMSDestination = msg.destination
+      INSERT JMSMessageType(myObj)
+  }
+```
+
+*   **NOTE**: Messages returned by a Select statement are in the exact same format as those sent to the source as a Notification.
+
+## Publish Statements
+
+In order to send messages to either a queue or a topic, a VAIL Publish statement must be used. The Publish statement will have
+three parameters: *message*, *queue/topic*, *JMSFormat*. The *message* parameter is simply the message to be sent. The next
+parameter is either *queue* or *topic*, depending on which type of destination you are sending the message to. The value of this
+parameter is the name of the given queue/topic. Finally, the last parameter is the *JMSFormat*, which specifies the JMS Message
+Type that will be used to send the message. If no custom Message Handler was implemented, the three values that we currently
+support are "TextMessage", "MapMessage", and "Message". The "Message" type does not have any message body, so no message parameter
+is needed. The following two examples show how to send all three supported message types.
+
+### Sending Text Message ###
+
+```
+PROCEDURE sendTextMessage()
+
+var msg = "A message sent from VANTIQ to a topic."
+var dest = "NamirJMSServer-0/NamirSystemModule-0!NamirJMSServer-0@/com/namir/weblogic/base/dt"
+var msgFormat = "TextMessage"
+
+PUBLISH {message: msg, topic: dest, JMSFormat: msgFormat} to SOURCE JMS1 
+```
+
+### Sending Map Message ###
+```
+PROCEDURE sendMapMessage()
+
+var msg = {}
+msg.age = 22
+msg.name = "Namir"
+var dest = "NamirJMSServer-0/NamirSystemModule-0!NamirJMSServer-0@/com/namir/weblogic/base/dq"
+var msgFormat = "MapMessage"
+
+PUBLISH {message: msg, queue: dest, JMSFormat: msgFormat} to SOURCE JMS1 
+```
+
+### Sending Message ###
+```
+PROCEDURE sendMapMessage()
+
+var dest = "NamirJMSServer-0/NamirSystemModule-0!NamirJMSServer-0@/com/namir/weblogic/base/dq"
+var msgFormat = "Message"
+
+PUBLISH {queue: dest, JMSFormat: msgFormat} to SOURCE JMS1 
+```
+
+## Custom Message Handlers ##
+
+To create custom Message Handler, you must implement the 
+```io.vantiq.extsrc.jmsSource.communication.messageHandler.MessageHandlerInterface.java``` interface. We recommend creating a
+subclass of the BaseMessageHandler class, and adding extra message formatting for the JMS Message Types that we do not 
+currently support. Once you have implemented the MessageHandlerInterface, you must add the class to the java classpath. If we 
```

Comment:
You will also need to ensure that any other classes used by your MessageHandler are available on the class path.